New package: kassiopeia

wdconinc opened the pull request:
Kassiopeia allows to run highly customizable particle tracking simulations along with calculations of electric and magnetic fields. 

Link: https://github.com/KATRIN-Experiment/Kassiopeia

(Part of an effort to contribute spack packages from the Electron-Ion Collider repository to the broader NP and HEP community.)

wdconinc committed:
[kassiopeia] New package

wdconinc committed:
[kassiopeia] Remove master branch, update dependencies

adamjstewart commented on var/spack/repos/builtin/packages/kassiopeia/package.py:
```diff
@@ -0,0 +1,57 @@
+    depends_on('zlib')
+    depends_on('root@6.0.0:', when='+root')
+    depends_on('vtk@6.1:', when='+vtk')
+    depends_on('openmpi', when='+mpi')
```
No other MPI framework is supported?

wdconinc replied:
Confirmed that this works with both openmpi and mpich.

adamjstewart replied:
In that case:
```suggestion
    depends_on('mpi', when='+mpi')
```

adamjstewart commented on var/spack/repos/builtin/packages/kassiopeia/package.py:
```diff
@@ -0,0 +1,57 @@
+    depends_on('root@6.0.0:', when='+root')
+    depends_on('vtk@6.1:', when='+vtk')
+    depends_on('openmpi', when='+mpi')
+    depends_on('intel-tbb', when='+tbb')
```
```suggestion
    depends_on('tbb', when='+tbb')
```
`tbb` is provided by `intel-tbb` and `intel-parallel-studio+tbb`

adamjstewart commented on var/spack/repos/builtin/packages/kassiopeia/package.py:
```diff
@@ -0,0 +1,57 @@
+    def cmake_args(self):
+        args = []
+        if self.spec.variants['vtk']:
+            args.extend(['-DKASPER_USE_VTK=ON'])
```
```suggestion
            args.append('-DKASPER_USE_VTK=ON')
```
Also, it would be nice to explicitly disable these in an else clause.

wdconinc committed:
Update var/spack/repos/builtin/packages/kassiopeia/package.py

Unable to test since I do not have a license to intel-parallel-studio, but I see no reason why it would not work if.

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

wdconinc committed:
[kassiopeia] depends_on mpi

Co-authored-by: Adam J. Stewart <ajstewart426@gmail.com>

wdconinc committed:
[kassiopeia] cmake_args with self.spec.satisfies and elses

adamjstewart commented on var/spack/repos/builtin/packages/kassiopeia/package.py:
```diff
@@ -0,0 +1,65 @@
+        if self.spec.satisfies('+opencl'):
+            args.extend(['-DKEMField_USE_OPENCL=ON'])
+        else:
+            args.extend(['-DKEMField_USE_OPENCL=OFF'])
```
```suggestion
            args.append('-DKEMField_USE_OPENCL=OFF')
```

wdconinc replied:
Yeah sorry. Brain fart. Not even sure why I used extend in the first place. Appreciate the patience.

wdconinc committed:
[kassiopeia] args.extend -> args.append
